watertap/unit_models/electroNP_ZO.py

```diff
@@ -273,25 +111,23 @@ def solute_treated_equation(b, t, p, j):
             units=pyunits.dimensionless,
         )
 
-        # NOTE: revisit if we should sum soluble nitrogen as total nitrogen
         @self.Constraint(
             self.flowsheet().time,
-            self.config.property_package.solute_set,
-            doc="Default solute removal equations",
-        )
-        def default_solute_removal_equation(b, t, j):
-            if j == "S_PO4":
-                return b.removal_frac_mass_comp[t, j] == b.P_removal
-            elif j == "S_NH4":
-                return b.removal_frac_mass_comp[t, j] == b.N_removal
+            self.config.property_package.component_list,
+            doc="soluble fraction",
+        )
+        def split_components(blk, t, i):
+            if i == "H2O":
+                return (
+                    blk.split_fraction[t, "byproduct", i]
+                    == 1 - blk.recovery_frac_mass_H2O[t]
+                )
+            elif i == "S_PO4":
+                return blk.split_fraction[t, "byproduct", i] == blk.P_removal
+            elif i == "S_NH4":
+                return blk.split_fraction[t, "byproduct", i] == blk.N_removal
             else:
-                return b.removal_frac_mass_comp[t, j] == 1e-7
-
-        self._stream_table_dict = {
-            "Inlet": self.inlet,
-            "Treated": self.treated,
-            "Byproduct": self.byproduct,
-        }
+                return blk.split_fraction[t, "byproduct", i] == 1e-7
```

Comment:
I did consider it, but I thought would be best to leave it as constraints in case we get expressions on how to calculate this through data. But I could change it